src/eventsub/channel/bits/use.rs

```diff
@@ -0,0 +1,128 @@
+#![doc(alias = "channel.bits.use")]
+//! sends a notification whenever Bits are used on a channel
+
+use super::*;
+
+/// [`channel.bits.use`](https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types/#channelbitsuse): sends a notification whenever Bits are used on a channel
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "typed-builder", derive(typed_builder::TypedBuilder))]
+#[cfg_attr(feature = "deny_unknown_fields", serde(deny_unknown_fields))]
+#[non_exhaustive]
+pub struct ChannelBitsUseV1 {
+    /// The ID of the broadcaster that you want to get Channel Bits notifications for.
+    #[cfg_attr(feature = "typed-builder", builder(setter(into)))]
+    pub broadcaster_user_id: types::UserId,
+}
+
+impl ChannelBitsUseV1 {
+    /// The ID of the broadcaster that you want to get Channel Bits notifications for.
+    pub fn broadcaster_user_id(broadcaster_user_id: impl Into<types::UserId>) -> Self {
+        Self {
+            broadcaster_user_id: broadcaster_user_id.into(),
+        }
+    }
+}
+
+impl EventSubscription for ChannelBitsUseV1 {
+    type Payload = ChannelBitsUseV1Payload;
+
+    const EVENT_TYPE: EventType = EventType::ChannelBitsUse;
+    #[cfg(feature = "twitch_oauth2")]
+    const SCOPE: twitch_oauth2::Validator =
+        twitch_oauth2::validator![any(twitch_oauth2::Scope::BitsRead)];
+    const VERSION: &'static str = "1";
+}
+
+/// [`channel.bits.use`](ChannelBitsUseV1) response payload.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "deny_unknown_fields", serde(deny_unknown_fields))]
+#[non_exhaustive]
+pub struct ChannelBitsUseV1Payload {
+    /// The User ID of the redeeming user.
+    pub user_id: types::UserId,
+    /// The login name of the redeeming user.
+    pub user_login: types::UserName,
+    /// The display name of the redeeming user.
+    pub user_name: types::DisplayName,
+    /// The User ID of the channel where the Bits were redeemed.
+    pub broadcaster_user_id: types::UserId,
+    /// The login of the channel where the Bits were used.
+    pub broadcaster_user_login: types::UserName,
+    /// The display name of the channel where the Bits were used.
+    pub broadcaster_user_name: types::DisplayName,
+    /// The number of Bits used.
+    pub bits: usize,
+    /// The type of Bits used.
+    #[serde(rename = "type")]
+    pub type_: BitsType,
+    /// Data about Power-up.
+    pub power_up: Option<BitsPowerUp>,
+    /// An object that contains the user message and emote information needed to recreate the message.
+    pub message: Option<crate::eventsub::channel::chat::Message>,
+}
```

Comment:
Hmm, I wonder now if it would make sense to make this into an [internally tagged enum](https://serde.rs/enum-representations.html#internally-tagged)

Reply:
I don't think that's possible, because of the surrounding junk. If anything, we'd need to do it like `action` in `channel.moderate` where we flatten the enum.

This doesn't generate nice code, though, because serde can't see the flattened type (it has to parse into some intermediate structure). I mentioned this somewhere - a custom (de)serialize impl is the only way to have an efficient implementation here (or we hope for reflection).